internal/harness/matchfunction/golang/apisrv/apisrv.go

```diff
@@ -0,0 +1,53 @@
+/*
+package apisrv provides an implementation of the gRPC server defined in
+../../../api/protobuf-spec/matchfunction.proto.
+
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package apisrv
+
+import (
+	"context"
+
+	"github.com/GoogleCloudPlatform/open-match/config"
+	"github.com/GoogleCloudPlatform/open-match/internal/pb"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// This is the function signature for the function to be implemented by the user.
+// The harness will pass the Rosters and PlayerPool for the match profile to this
+// function and it will return the Rosters to be populated in the proposal.
+type MatchFunction func(context.Context, string, []*pb.Roster, []*pb.PlayerPool) (string, []*pb.Roster, error)
+
+// MatchFunctionServer implements pb.MatchFunctionServer, the server generated
+// by compiling the protobuf, by fulfilling the pb.MatchFunctionServer interface.
+type MatchFunctionServer struct {
```

Comment:
What's the difference between this an OpenMatchServer? Couldn't we just use the standard grpc binding stuff here and add a wrapper for MatchFunction?

Reply:
This does not need redis pool initialization - also this one needs to pass the MatchFunction to the server. I started with using the OpenMatchServer but mucking with its internals to add things specific to harness didnt seem right. This could possibly be refactored back into serving package as a HarnessServer separating out the construction of the params and sharing some code between the types of servers - may be for the future refactor in 0.6